agent/api/task/task.go

```diff
@@ -325,7 +325,7 @@ func (task *Task) initializeVolumes(cfg *config.Config, dockerClient dockerapi.D
 // able to handle such an occurrence appropriately (e.g. behave idempotently).
 func (task *Task) PostUnmarshalTask(cfg *config.Config,
 	credentialsManager credentials.Manager, resourceFields *taskresource.ResourceFields,
-	dockerClient dockerapi.DockerClient, ctx context.Context) error {
+	dockerClient dockerapi.DockerClient, ctx context.Context, options ...Option) error {
```

Comment:
I like the idea, but call to `PostUnmarshalTask` might become cumbersome with too many options

Reply:
It could, yes. I can't think of a better alternative that doesn't involve adding unnecessary complexity to `task` though. (like it is before this refactor).
The whole `PostUnmarshalTask` is already cumbersome , IMHO.

Is there an alternative there?

Reply:
if what worries you is that we might end up in the future with a call like:

```golang
task.PostUnmarshalTask(cfg, credentialsManager, resourceFields, dockerClient, ctx,
opt1,
opt2,
opt3,
opt4,
opt5)
```

We can always create an option like:

```golang
func compositeOpt(task *Task) {
    opt1(task)
    opt2(task)
    opt3(task)
    opt4(task)
    opt5(task)
}
```

And then just call `PostUnmarshalTask` like this:

```golang
task.PostUnmarshalTask(cfg, credentialsManager, resourceFields, dockerClient, ctx, compositeOpt)
```

Reply:
ya, we could refactor later if needed, I can't think of a better alternative 😅